torch/_inductor/codegen/triton.py

```diff
@@ -2643,11 +2673,11 @@ def decide_later():
                 dtype = torch.bool
 
```

Comment:
Just curious - did you try measuring overhead of gdc_wait in loop ? 

e.g. putting it here: 

https://gist.github.com/v0i0/ce830bbaa651041e3ceef6fe0e743f46#file-output_code-log-L170

where we are executing (slightly more) kernel prologue setup. 

This is maybe less relevant in current pr but more relevant if it allows us to get an extra load in.